base/docker/scripts/status.py

```diff
@@ -1,73 +1,21 @@
-from typing import List, Optional
+from typing import List
 from prometheus_client import Gauge, Enum, Info
 import logging
-from enum import Enum as PyEnum
-from typer import Typer
-import os
-import requests
+from status_tools import WorkFlowError
 
 logger = logging.getLogger(__name__)
 
-class WorkflowStatus(PyEnum):
-    STARTING = "starting"
-    RUNNING = "running"
-    STOPPING = "completed"
-    FAILED = "failed"
 
-class WorkFlowError(PyEnum):
-    DB_ERROR = "db_error"
-    WORKFLOW_ERROR = "workflow_error"
-    API_ERROR = "api_error"
 
-def singleton(cls):
-    instances = {}
-    def get_instance(*args, **kwargs):
-        if cls not in instances:
-            instances[cls] = cls(*args, **kwargs)
-        return instances[cls]
-    return get_instance
-
-class StatusUpdater:
-    def post_update(
-            self,
-            completed: Optional[float] = None,
-            phases: Optional[List[str]] = None,
-            phase: Optional[str] = None,
-            error_message: Optional[str] = None,
-            error_code: Optional[WorkFlowError] = None,
-            status: Optional[WorkflowStatus] = None,
-            accessible: Optional[bool] = None):
-        wf_status = {
-        }
-        if completed is not None:
-            wf_status["completeness"] = float(completed) / 100.0
-        if phases is not None:
-            wf_status["phases"] = phases
-        if phase is not None:
-            wf_status["phase"] = phase
-        if error_message is not None:
-            wf_status["error_message"] = error_message
-        if error_code is not None:
-            wf_status["error_code"] = error_code.value
-        if status is not None:
-            wf_status["status"] = status.value
-        else:
-            wf_status["status"] = WorkflowStatus.RUNNING.value
-        if accessible is not None:
-            wf_status["accessible"] = accessible
-        if len(wf_status) > 0:
-            print(f"Updating status: {wf_status}")
-            try:
-                update_response = requests.post(
-                        f"http://{os.environ.get('HOSTNAME')}:8080/response",
-                        json=wf_status
-                    )
-            except Exception as e:
-                print(f"Failed to update status: {e}")
+class Singleton(type):
```

Comment:
I'm uncomfortable with a Singleton implementation that accepts construction arguments, but doesn't either use them as cache keys or check that they're the same.